File: banking/src/main/java/com/bank/service/AccountService.java
package com.bank.service;

import com.bank.entity.Account;
import com.bank.entity.AccountType;

import java.math.BigDecimal;
import java.util.List;

public interface AccountService {
    Account createAccount(AccountType type, String email);
    List<Account> getMyAccounts(String email);
    Account getAccount(Long id, String email);
    void transferByAccountNumber(
            Long fromAccountId,
            String toAccountNumber,
            String toCustomerId,
            BigDecimal amount,
            String description,
            String senderEmail
    );
    
}

File: frontend/bank/src/app/account/account.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class AccountService {

  private baseUrl = '/api';

  constructor(private http: HttpClient) {}

  getAccounts(): Observable<any[]> {
    return this.http.get<any[]>(`${this.baseUrl}/accounts`);
  }

  createAccount(accountType: string): Observable<any> {
    return this.http.post<any>(
      `${this.baseUrl}/accounts?type=${accountType}`,
      {}
    );
  }

  getAccount(id: number): Observable<any> {
    return this.http.get<any>(`${this.baseUrl}/accounts/${id}`);
  }

  transact(accountId: number, transactionData: any): Observable<any> {
    return this.http.post<any>(
      `${this.baseUrl}/accounts/${accountId}/transact`,
      transactionData
    );
  }

  getTransactions(accountId: number): Observable<any[]> {
    return this.http.get<any[]>(
      `${this.baseUrl}/accounts/${accountId}/transactions`
    );
  }

  // ✅ ADD THIS: ACCOUNT → ACCOUNT TRANSFER
  transferMoney(payload: {
    fromAccountId: number;
    toAccountNumber: string;
    toCustomerId: string;
    amount: number;
    description?: string;
  }): Observable<any> {
    return this.http.post<any>(
      `${this.baseUrl}/accounts/transfer`,
      payload
    );
  }
}